### ISSUE TITLE
[Mac] Why are bitmaps always drawn in integral mode?

### ISSUE THREAD

**Issue by geraintluff** [steinbergmedia/vstgui#90]
On Mac the bitmap drawing methods in `CGDrawContext.cpp` always use integral mode:

https://github.com/steinbergmedia/vstgui/blob/2bf29afb1a6cadae2f189e8266a53d3b96d2ff74/vstgui/lib/platform/mac/cgdrawcontext.cpp#L609

https://github.com/steinbergmedia/vstgui/blob/2bf29afb1a6cadae2f189e8266a53d3b96d2ff74/vstgui/lib/platform/mac/cgdrawcontext.cpp#L573

Every other invocation of `beginContext()` in that file uses the integral mode as the second argument, looking like this:

https://github.com/steinbergmedia/vstgui/blob/2bf29afb1a6cadae2f189e8266a53d3b96d2ff74/vstgui/lib/platform/mac/cgdrawcontext.cpp#L136

I found this out because I was drawing a rotating dial.  The rotation works fine (including interpolation), but the offset is rounded to the nearest integral position, causing my dial to jump around within a pixel [(screencap video)](https://star-catapult.net/public/share/dial-rotation-jumping-around.mp4).

_(I haven't checked this on Windows (only just starting out), so I don't know whether `gdiplusdrawcontext.cpp` does the same thing.)_

**Comment by scheffle** [comment]
Yes, this is intentional. Otherwise bitmaps will be drawn blurry when the scale factor of the transformation matrix is not an integer.
Why don't you pre render your dial and use the CAnimKnob class ?

**Comment by geraintluff** [comment]
If the transformation matrix is doing non-integer scaling, then your bitmap's location might land on a non-integer co-ordinate, yes.  It's likely to be scaling/interpolating the bitmap anyway though - I certainly didn't see a visual difference when I changed it, except my offsets stopped being rounded.

Anyway, If that's the issue, couldn't users enable/disable integral mode for that draw?

The same issue happens when you draw a line which is meant to be exact-pixel-aligned, but a transform means it's not - I though choosing what to do in that situation is exactly what integral mode is for?

**Comment by scheffle** [comment]
You'll see the difference when you use a CNinePartTiledBitmap on a scaled context.
And for line drawing, pixel aligned drawing should work with a transform, sure only the start and end point is pixel aligned, but if you choose a line width of exactly one pixel (see CDrawContext::getHairlineSize) then it is pixel aligned.
You've not answered my question, why you don't choose a pre rendered bitmap ?
But anyway, to add an option to draw a bitmap not pixel aligned is always an option. If you like you can work this out and create a pull request.
